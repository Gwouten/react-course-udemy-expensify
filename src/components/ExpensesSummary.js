import React from 'react';
import { connect } from 'react-redux';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
  <div>
    <p>Viewing {props.expenseCount} {props.expenseCount === 1 ? 'expense' : 'expenses'} totalling {numeral(props.expensesTotal/100).format('$0,0.00')}</p>
  </div>
);

const mapStateToProps = (state, props) => ({
  expenseCount: state.expenses.length,
  expensesTotal: expensesTotal(state.expenses)
});

export default connect(mapStateToProps)(ExpensesSummary);
