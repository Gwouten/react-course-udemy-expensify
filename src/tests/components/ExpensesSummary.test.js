import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should generate snapshot for 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary
    expenseCount={1}
    expensesTotal={235}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should generate snapshot for multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary
    expenseCount={23} 
    expensesTotal={254478425}
  />);
  expect(wrapper).toMatchSnapshot();
});
