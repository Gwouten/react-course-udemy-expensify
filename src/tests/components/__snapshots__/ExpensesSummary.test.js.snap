// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate snapshot for 1 expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h3>
</div>
`;

exports[`should generate snapshot for multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    23
     
    expenses
     totalling 
    $2,544,784.25
  </h3>
</div>
`;
